codebuild_build.sh

```diff
@@ -0,0 +1,193 @@
+    echo "               * Use additional (-s) in <sourceIdentifier>:<sourceLocation> format for secondary source"
+    echo "               * For sourceIdentifier, use a value that is fewer than 128 characters and contains only alphanumeric characters and underscores"
+    echo "  -c        Use the AWS configuration and credentials from your local host. This includes ~/.aws and any AWS_* environment variables."
+    echo "  -p        Used to specify the AWS CLI Profile."
+    echo "  -b FILE   Used to specify a buildspec override file. Defaults to buildspec.yml in the source directory."
+    echo "  -m        Used to mount the source directory to the customer build container directly."
+    echo "  -d        Used to run the build container in docker privileged mode."
+    echo "  -e FILE   Used to specify a file containing environment variables."
+    echo "            (-e) File format expectations:"
+    echo "               * Each line is in VAR=VAL format"
+    echo "               * Lines beginning with # are processed as comments and ignored"
+    echo "               * Blank lines are ignored"
+    echo "               * File can be of type .env or .txt"
+    echo "               * There is no special handling of quotation marks, meaning they will be part of the VAL"
+    exit 1
+}
+
+image_flag=false
+artifact_flag=false
+awsconfig_flag=false
+mount_src_dir_flag=false
+docker_privileged_mode_flag=false
+
+while getopts "cmdi:a:s:b:e:l:p:h" opt; do
+    case $opt in
+        i  ) image_flag=true; image_name=$OPTARG;;
+        a  ) artifact_flag=true; artifact_dir=$OPTARG;;
+        b  ) buildspec=$OPTARG;;
+        c  ) awsconfig_flag=true;;
+        m  ) mount_src_dir_flag=true;;
+        d  ) docker_privileged_mode_flag=true;;
+        s  ) source_dirs+=("$OPTARG");;
+        e  ) environment_variable_file=$OPTARG;;
+        l  ) local_agent_image=$OPTARG;;
+        p  ) aws_profile=$OPTARG;;
+        h  ) usage; exit;;
+        \? ) echo "Unknown option: -$OPTARG" >&2; exit 1;;
+        :  ) echo "Missing option argument for -$OPTARG" >&2; exit 1;;
+        *  ) echo "Invalid option: -$OPTARG" >&2; exit 1;;
+    esac
+done
+
+if  ! $image_flag
+then
+    echo "The image name flag (-i) must be included for a build to run" >&2
+fi
+
+if  ! $artifact_flag
+then
+    echo "The artifact directory (-a) must be included for a build to run" >&2
+fi
+
+if  ! $image_flag ||  ! $artifact_flag
+then
+    exit 1
+fi
+
+docker_command="docker run -it "
+if isOSWindows
+then
+    docker_command+="-v //var/run/docker.sock:/var/run/docker.sock -e "
+else
+    docker_command+="-v /var/run/docker.sock:/var/run/docker.sock -e "
+fi
+
+docker_command+="\"IMAGE_NAME=$image_name\" -e \
+    \"ARTIFACTS=$(allOSRealPath "$artifact_dir")\""
+
+if [ -z "$source_dirs" ]
+then
+    docker_command+=" -e \"SOURCE=$(allOSRealPath "$PWD")\""
+else
+    for index in "${!source_dirs[@]}"; do
+        if [ $index -eq 0 ]
```

Comment:
Codacy found an issue: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/gh/EddieDemoOrg/petstore/pullRequest?prid=6129301)